src/Tasks/Microsoft.Common.CurrentVersion.targets

```diff
@@ -325,7 +325,7 @@ Copyright (C) Microsoft Corporation. All rights reserved.
     <TargetPath Condition=" '$(TargetPath)' == '' ">$(TargetDir)$(TargetFileName)</TargetPath>
 
     <TargetRefPath Condition=" '$(TargetRefPath)' == '' and '$(ProduceReferenceAssembly)' == 'true' and ('$(ProduceReferenceAssemblyInOutDir)' == 'true' or '$([MSBuild]::AreFeaturesEnabled(17.0))' != 'true' ) ">$([MSBuild]::NormalizePath($(TargetDir), 'ref', $(TargetFileName)))</TargetRefPath>
-    <TargetRefPath Condition=" '$(TargetRefPath)' == '' and '$(ProduceReferenceAssembly)' == 'true' ">$([MSBuild]::NormalizePath($(IntermediateOutputPath), 'ref', $(TargetFileName)))</TargetRefPath>
+    <TargetRefPath Condition=" '$(TargetRefPath)' == '' and '$(ProduceReferenceAssembly)' == 'true' ">$([MSBuild]::NormalizePath($(MSBuildProjectDirectory), $(IntermediateOutputPath), 'ref', $(TargetFileName)))</TargetRefPath>
```

Comment:
Is there validation that `$(IntermediateOutputPath)` is not an absolute path here?

Reply:
Yes, it is defined here:
https://github.com/dotnet/msbuild/blob/6dba77a45e3af8fe3058040b870bd5a5eb045962/src/Tasks/Microsoft.Common.CurrentVersion.targets#L159-L163

The default `$(IntermediateOutputPath)` ends up being something like `obj\Debug` or `obj\x64\Debug`.

Reply:
That would still mean that users could make this fail with something like `msbuild myProj.csproj /p:IntermediateOutputPath=C:\foo\bar`, right?

Reply:
Probably not. It calls `Path.Combine` internally which behaves nicely:
![image](https://user-images.githubusercontent.com/25249058/126610104-6b19029c-a030-4648-b14f-794c4a02f88e.png)